src/components/ScreenStack.tsx

```diff
@@ -77,6 +77,12 @@ function ScreenStack(props: ScreenStackProps) {
     },
   });
 
+  const preloadedScreensCount = React.Children.toArray(children).filter(
+    // @ts-expect-error it's either SceneView in v6 or RouteView in v5
+    child => child.props.isPreloaded,
+  ).length;
```

Comment:
I think it's a bit fragile to read it like this, and can break if we reorganize our components. It'd be better to have something more explicit that can be type checked.

Reply:
Yeah, it was done quickly to check if the logic is correct. Still, we get the direct children of `ScreenStack`, so how would you see it done better?

Reply:
@WoLewicki, it may not always be the direct child in how `React.Children` sees it. `React.Children` only sees it if you have written it as a direct child, but it could also be a direct child in terms of the React tree, but in a sub component.

in addition, this also relies on a prop called `isPreloaded` - an internal name, passed to an internal component `SceneView`. any refactor that changes how `isPreloaded` is determined or passed, or any change to the name of the prop or component signature can break this code.

using `React.Children` is only ok if you control the code that you access via the children API, or if your library's API already enforces this (like `Stack.Navigator` requires `Stack.Screen` as direct children with specific props passed to it).

since `ScreenStack`'s public API doesn't enforce what direct child its direct child is, and whether it should have an `isPreloaded` prop or not, accessing children here will always be fragile.